model_api/python/model_api/models/action_classification.py

```diff
@@ -0,0 +1,220 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from model_api.adapters.utils import RESIZE_TYPES, InputTransform
+
+from .model import Model
+from .types import BooleanValue, ListValue, NumericalValue, StringValue
+from .utils import ClassificationResult, load_labels
+
+if TYPE_CHECKING:
+    from model_api.adapters.inference_adapter import InferenceAdapter
+
+
+class ActionClassificationModel(Model):
+    """An wrapper for an action classification model
+
+    The ActionClassificationModel has 1 or more inputs with images - 6D tensors.
+    It may support additional inputs - 4D tensors.
+    Two input formats are supported. One is 'NSCTHW' and another one is 'NSTHWC'.
+    What each alphabet means are as below.
+    N => batch size / S => number of clips x number of crops / C => number of channels
+    T => time / H => height / W => width
+
+    Video format is different from image format, so OpenVINO PrePostProcessors isn't available.
+    For that reason, basic preprocessing such as resize and normalize are conducted in this class.
+
+    Attributes:
+        image_blob_names (List[str]): names of all image-like inputs (6D tensors)
+        image_info_blob_names (List[str]): names of all secondary inputs (4D tensors)
+        image_blob_name (str): name of the first image input
+        resize_type (str): the type for image resizing (see `RESIZE_TYPE` for info)
+        resize (function): resizing function corresponding to the `resize_type`
+        input_transform (InputTransform): instance of the `InputTransform` for image normalization
+    """
+
+    __model__ = "Action Classification"
+
+    def __init__(
+        self,
+        inference_adapter: InferenceAdapter,
+        configuration: dict[str, Any] = dict(),
+        preload: bool = False,
+    ) -> None:
+        """Action classaification model constructor
+
+        Args:
+            inference_adapter (InferenceAdapter): allows working with the specified executor
+            configuration (dict, optional): it contains values for parameters accepted by specific
+              wrapper (`labels` `mean_values`, etc.) which are set as data attributes
+            preload (bool, optional): a flag whether the model is loaded to device while
+              initialization. If `preload=False`, the model must be loaded via `load` method before inference
+
+        Raises:
+            WrapperError: if the wrapper failed to define appropriate inputs for images
+        """
+        super().__init__(inference_adapter, configuration, preload)
+        self.image_blob_names, self.image_info_blob_names = self._get_inputs()
```

Comment:
Is `self.image_info_blob_names` used somewhere?

Reply:
It isn't. I'll remove it.

Reply:
done https://github.com/openvinotoolkit/model_api/pull/178/commits/7e4dd83d40c20d88ab7dbe19f42315cb2e21dc2d